frontend/src/pages/projects/notebook/utils.ts

```diff
@@ -145,40 +145,56 @@ export const useNotebookStatus = (
   let status: EventStatus = EventStatus.IN_PROGRESS;
   const lastItem = filteredEvents[filteredEvents.length - 1];
   let currentEvent = '';
-  if (lastItem.message.includes('oauth-proxy')) {
-    switch (lastItem.reason) {
-      case 'Pulling': {
-        currentEvent = 'Pulling oauth proxy';
-        percentile = 72;
-        break;
-      }
-      case 'Pulled': {
-        currentEvent = 'Oauth proxy pulled';
-        percentile = 80;
-        break;
-      }
-      case 'Created': {
-        currentEvent = 'Oauth proxy container created';
-        percentile = 88;
-        break;
-      }
-      case 'Started': {
-        currentEvent = 'Oauth proxy container started';
-        percentile = 96;
-        break;
-      }
-      case 'Killing': {
-        currentEvent = 'Stopping container oauth-proxy';
-        status = EventStatus.WARNING;
-        break;
-      }
-      default: {
-        if (lastItem.type === 'Warning') {
-          currentEvent = 'Issue creating oauth proxy container';
-          status = EventStatus.WARNING;
+
+  function handleProxyEvent(proxyType: string, event: EventKind) {
```

Comment:
```suggestion
  function handleProxyEvent(proxyType: 'istio' | 'oauth', event: EventKind) {
```
Let us just make it more prescriptive what types are possible for sanity more than anything else.